keel-core-test/src/main/kotlin/com/netflix/spinnaker/keel/persistence/ArtifactRepositoryTests.kt

```diff
@@ -492,17 +493,62 @@ abstract class ArtifactRepositoryTests<T : ArtifactRepository> : JUnit5Minutests
             .isNotEqualTo(pin1.version)
         }
 
-        test("latestVersionApprovedIn prefers a pinned version over the latest approved version") {
-          subject.pinEnvironment(manifest, pin1)
-          expectThat(subject.latestVersionApprovedIn(manifest, artifact2, environment2.name))
-            .isEqualTo(version4)
-            .isEqualTo(pin1.version)
+        test("get env artifact version shows that artifact is not pinned") {
+          val envArtifactSummary = subject.getArtifactSummaryInEnvironment(
+            deliveryConfig = manifest,
+            environmentName = pin1.targetEnvironment,
+            artifactReference = artifact2.reference,
+            version = version4
+          )
+          expectThat(envArtifactSummary)
+            .isNotNull()
+            .get { pinned }
+            .isFalse()
         }
 
-        test("pinned version cannot be vetoed") {
-          subject.pinEnvironment(manifest, pin1)
-          expectThat(subject.markAsVetoedIn(manifest, artifact2, pin1.version!!, pin1.targetEnvironment))
-            .isFalse()
+        context("once pinned") {
+          before {
+            subject.pinEnvironment(manifest, pin1)
+          }
+
+          test("latestVersionApprovedIn prefers a pinned version over the latest approved version") {
+            expectThat(subject.latestVersionApprovedIn(manifest, artifact2, environment2.name))
+              .isEqualTo(version4)
+              .isEqualTo(pin1.version)
+          }
+
+          test("pinned version cannot be vetoed") {
+            expectThat(subject.markAsVetoedIn(manifest, artifact2, pin1.version!!, pin1.targetEnvironment))
```

Comment:
I was surprised `EnvironmentArtifactPin.version` could be `null`, since the whole point is to pin a specific version. Do you know if that was intentional? Might be worth fixing in this PR if not.

Reply:
We throw an exception in the controller if it's null. So, I don't know why it can be null in the first place. I might look into it now, good suggestion.